akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -367,6 +383,24 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
   }
 
   /**
+   * Java API: The supplied thunk will be run, once, when current cluster member is `Removed`. Used w/ error exit.
+   * If the cluster has already been shutdown the thunk will run on the caller thread immediately.
+   * Typically used together `cluster.leave(cluster.selfAddress)` and then `system.terminate()`.
+   */
+  def registerOnMemberRemovedAfterDown(callback: Runnable): Unit = {
+    registerOnMemberRemoved(callback)
+  }
+
+  /**
+   * Java API: The supplied thunk will be run, once, when current cluster member is `Removed`. Used w/ graceful exit.
+   * If the cluster has already been shutdown the thunk will run on the caller thread immediately.
+   * Typically used together `cluster.leave(cluster.selfAddress)` and then `system.terminate()`.
+   */
+  def registerOnMemberRemovedAfterLeaving(callback: Runnable): Unit = {
+    registerOnMemberRemoved(callback)
```

Comment:
Maybe it makes sense to revisit this once the work in #21930 is complete since it seems that may render this unnecessary. My interpretation of the original issue was that just having additional callbacks could make the triggered events clearer.